### PR TITLE
start dlv on free port

### PR DESCRIPTION
**Pull request by beardnick**
start dlv on free port every time so that you can debug multiple go projects at the same time.

### CODE REVIEW

File: lua/dap-go.lua [hunk 1]
```diff
@@ -29,6 +30,21 @@ local function load_module(module_name)
...
+  local tcp = uv.new_tcp()
+  tcp:bind('127.0.0.1', 0)
+  local port = tcp:getsockname().port
+  tcp:shutdown()
+  tcp:close()
+  return port
+end
+
+local function get_addr(host,port)
+    port = get_free_port()
```
Comment (leoluz) [inline lua/dap-go.lua:43]: Will this honour pre-configured port? If users define the port, it should take precedence over the random logic.

**Comment by beardnick** [comment]
> I like the new simplified version better. Please take a look at my comment. Also, do you mind adding a brief explanation in the readme file about how how/when the new random port logic works please?

Should I add the explanation to the feature 1? 
Something like this
Auto launch Delve on free port. No configuration needed. You just have to have dlv in your path.

**Comment by leoluz** [comment]
> > I like the new simplified version better. Please take a look at my comment. Also, do you mind adding a brief explanation in the readme file about how how/when the new random port logic works please?
> 
> Should I add the explanation to the feature 1? Something like this Auto launch Delve on free port. No configuration needed. You just have to have dlv in your path.

Basically explaining that by default it will try to start delve in an ephemeral port but if necessary the port can be specified in the configuration to force a fixed number (ideally with an example).

**Comment by leoluz** [comment]
Start dlv on a free port is now natively supported by nvim-dap and I will implement the necessary code change as part of https://github.com/leoluz/nvim-dap-go/pull/26

Closing this for now.

**Pull request closed without merging**